### DISCUSSION TITLE
CF Roadmap: Time

### DISCUSSION THREAD

**Discussion by feggleton** [cf-convention/discuss#393] [Comments and ideas for changing CF]
### Topic for discussion

**Topic for Discussion**

At this year's CF Workshop [we divided the roadmap preparations into multiple themes](https://docs.google.com/document/d/1SzNCgMS2BfErOc9K26D-l9wKhVyRhhgq-UwbXx9fIiU/edit?tab=t.0). Time was suggested as something we should discuss in terms of the future direction in CF. After a talk from Chris L (UKMO), there were some questions about time and how it is represented in CF. These were:

- Should we have other ways of representing time than are provided by UDunits syntax?
- How easy is it to map from UDunits time/calendars to other standards?
- Can we represent a wider range of non-standard calendars than currently? E.g. paleoclimate, non-earth based calendars?
- Do we want to separate time units and units of measure? (I.e. handle them with different software libraries?)

Although I am not personally leading this topic in the roadmap publication, I am just starting the discussion for now.

**Comment by chris-little** [comment]
@feggleton My first thoughts in response to your questions:
I see no reason to have time units separate from other UoMs. That said, (SI) seconds are reliable, but (calendar) days, weeks, months and years are all variable in length at some precision. 
Calendars and coordinate reference systems best treated as different things.
There are plenty of conventional CRSs/UoM for time that are reasonably well defined: Millions of years ago, Unix time, Julian years and Julian days, with the advantageous characteristic of only one UoM for the time axis. Though Unix time and POSIX time have some ambiguities that most people are oblivious too. 

I think CF support for non-Earth calendars and CRSs should be on the roadmap. A Mars calendar with Martian years and Martian days (called Sols) has been well used.  And NASA has called for a Coordinated Lunar Time (atomic).

Notation to indicate which CRS or Calendar has been used is needed, preferably not just a CF notation, but something with the ubiquitousness of RFC3339 or ISO 8601.

I still propose to register T+hh type CRSs with OGC.

**Reply by feggleton** [reply]
@japamment wrote the questions above so she might want to comment on this.

**Comment by davidhassell** [comment]
Hello,

The second half (24-25 September) of the [2025 CF workshop](https://cfconventions.org/Meetings/2025-Workshop.html) is dedicated to the CF roadmap, including hackathons dedicated to advancing the road map topics. These hackathons have no pre-determined structure and can be used for anything from further debate to writing an actual proposal - whatever would be most useful the topic at this time.

If there is interest in working on this roadmap topic in this way, please let us know in a comment on this discussion soon, so that we can advertise it on the meeting agenda - this is a good opportunity to possibly bring in fresh ideas from people who may not yet have engaged with the roadmap.

Thanks, on behalf of the workshop committee.

**Reply by feggleton** [reply]
I don't know how much further discussion is needed on this topic at the moment but once a draft roadmap document is available, I think it will be worth writing something up for 'what CF might do with Time in the future' or 'the future of Time in CF' (no pun intended). Then I can engage Chris with that and also our space weather folks at the Met Office who are interested in using CF and would like to comment on the use of non-terrestrial 'time'. I started this Discussion but it is not something I have expertise on so just here to facilitate.

**Comment by chris-little** [comment]
@davidhassell If there is a workshop topic, I think my comment of 2024-12-04 above covers it. Plenty of detail could be added.

**Comment by pvanlaake** [comment]
Discussion #400 is coming to a close, with proposed text for a revised section 4.4 Time Coordinate for inclusion in CF 1.13. That should make a worthy workshop topic and I am happy to support that.

It is interesting to note that adding flexibility to section 4.4 for non-terrestrial time-keeping was discussed but put aside for the moment as there are more than a few things to consider here. Including that in the road map for time could be the first step forward.
